tests/bluetooth/host/buf/bt_buf_get_cmd_complete/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+if (NOT BOARD STREQUAL unit_testing)
+  message(FATAL_ERROR "This project can only be used with '-DBOARD=unit_testing'.")
+endif()
+
+FILE(GLOB SOURCES src/*.c)
+
+project(bluetooth_bt_buf_get_cmd_complete)
+
+find_package(Zephyr COMPONENTS unittest HINTS $ENV{ZEPHYR_BASE})
+
+include(${ZEPHYR_BASE}/tests/bluetooth/host/host_mocks.cmake)
```

Comment:
I think you already tried using a library and had a bunch of issues. Could you try again with torstein's changes to see if that resolves the issues?

Reply:
Yes, it's on my to-do list

Reply:
yeah I think at this point, @tejlmand is the one that can help.

Reply:
With some help from **Torsten**, I was able to bypass that issue and explicitly add the `autoconf.h` while building the library.
That is done using
`zephyr_library_compile_options(-imacros ${AUTOCONF_H})`

But currently, I'm encountering a different problem.
The build process is now terminated with `fatal error: kobj-types-enum.h: No such file or directory`.
This file is automatically generated under the build directory and I can find it.
I already added the expected path of that file using `zephyr_library_include_directories(${CMAKE_BINARY_DIR}/zephyr/include/generated)`.

I think that the file is generated at some point after trying to build the library or building the library requires that all files are there before starting the build process.

Any suggestions?

Reply:
please find the rough framework here:
https://github.com/ahmedmoheb-nordic/zephyr/pull/5